## Семинар 6

1. Известно, что генеральная совокупность распределена нормально, со средним квадратическим отклонением равным 16. Найти доверительный интервал для оценки математического ожидания с надежностью $0.95$, если выборочная средняя $M=80$, а объем выборки $n=256$.

Поскольку нам известно среднее квадратическое отклонение генеральной совокупности $\sigma=16$, то для расчета $95\%$-го доверительного интервала, будем использовать $Z$-критерий. И границы доверительного интервала расчитаем по формуле:  
$\displaystyle\bar X\pm z_{\alpha/2}\cdot\frac{\sigma}{\sqrt n}$, где:  
$\bar X=M=80$-среднее значение выборки    
$z_{\alpha/2}$ табличное значение $Z$-критерия, и для $95\%$-го доверительного интервала $z_{\alpha/2}=1,96$  
$\sigma=16$-среднее квадратическое отклонение генеральной совокупности  
$n=256$-объем выборки  
Таким образом $95\%$-й доверительный интервал расчитаем как: $\left[80-1.96\cdot\frac{16}{\sqrt 256};80+1.96\cdot\frac{16}{\sqrt 256}\right]$

In [1]:
left=80-1.96*16/256**(1/2)
right=80+1.96*16/256**(1/2)
print(f'95%-й доверительный интервал для оценки мат. ожидания генеральной совокупности: [{left};{right}].')

95%-й доверительный интервал для оценки мат. ожидания генеральной совокупности: [78.04;81.96].


---  
#### Ответ:  
$\left[78.04;81.96\right]$  

---

2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

Поскольку из условия задачи нам ничего не известно ни о мат. ожидании генеральной совокупности ни о среднем квадратическом отклонении для генеральной совокупности, то для расчета $95\%$-го доверительного интервала будем использовать $t$-критерий,  и формулу:  
$\displaystyle\bar X\pm t_{\alpha/2}\cdot\frac{\sigma_{несмещ}}{\sqrt{n}}$, где  
$\sigma_{несмещ}$-несмещённое среднее квадратическое отклонение выборки,  
$n$-размер выборки  
Воспользуемся встроенными функциями библиотеки numpy:

In [2]:
import numpy as np

In [3]:
arr=np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
print(f'Среднее выборочное: {np.mean(arr): .2f},\n'
      f'Размер выборки n={len(arr)},\n'
      f'Среднее квадратическое отклонение по выборке(несмещенное): {np.std(arr, ddof=1): .2f}.'
     )

Среднее выборочное:  6.59,
Размер выборки n=10,
Среднее квадратическое отклонение по выборке(несмещенное):  0.45.


Для вычисления $t$-критерия воспользуемся библиотекой scipy

In [4]:
import scipy.stats as stats

In [5]:
def t_from_table(confidens, len_array):
    alpha=(1-confidens)
    return stats.t.ppf(1-alpha/2, len_array-1)
print(f'Табличное значение t-критерия для 95%-го доверительного интервала данной выборки: {t_from_table(0.95, len(arr)): .3f}')

Табличное значение t-критерия для 95%-го доверительного интервала данной выборки:  2.262


Итоговая функция для расчета доверительного интервала:

In [6]:
def confidens_int(arr, confidens):
    return round(np.mean(arr)-t_from_table(confidens,len(arr))*np.std(arr, ddof=1)/len(arr)**0.5,3), \
           round(np.mean(arr)+t_from_table(confidens,len(arr))*np.std(arr, ddof=1)/len(arr)**0.5,3)

print(f'95%-й доверительный интервал для истинного значения Х: {confidens_int(arr, 0.95)}.')

95%-й доверительный интервал для истинного значения Х: (6.268, 6.912).


---  
#### Ответ:  
$\left[6.268;6.912\right]$  

---

3. Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170  
Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175  
Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.  

In [7]:
a = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
b = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])

In [8]:
x_1 = np.mean(a)
x_2 = np.mean(b)
delta = x_1 - x_2

In [9]:
n = len(a)
D1 = np.var(a, ddof = 1)
D2 = np.var(b, ddof = 1)

D = (D1 + D2) / 2
SE = np.sqrt(D/n + D/n)

t = stats.t.ppf(0.975, 2 * (n - 1))

result = (delta - t*SE, delta + t*SE)

In [11]:
print(f'95%-й доверительный интервал: {result}')

95%-й доверительный интервал: (-10.068418034506857, 6.268418034506846)


Ответ:

$\left[-10.068418034506857; 6.268418034506846\right]$  